lib/grandpa/message.go

```diff
@@ -77,21 +77,57 @@ func (v *VoteMessage) ToConsensusMessage() (*ConsensusMessage, error) {
 	}, nil
 }
 
-// NeighbourMessage represents a network-level neighbour message
-type NeighbourMessage struct {
-	Version byte
-	Round   uint64
-	SetID   uint64
-	Number  uint32
+// VersionedNeighbourPacket represents the enum of neighbour messages
+type VersionedNeighbourPacket scale.VaryingDataType
+
+// Index Returns VDT index
+func (VersionedNeighbourPacket) Index() uint { return 2 }
+
+func newVersionedNeighbourPacket() VersionedNeighbourPacket {
+	vdt := scale.MustNewVaryingDataType(NeighbourPacketV1{})
+
+	return VersionedNeighbourPacket(vdt)
+}
+
+// Set updates the current VDT value to be `val`
+func (vnp *VersionedNeighbourPacket) Set(val scale.VaryingDataTypeValue) (err error) {
+	vdt := scale.VaryingDataType(*vnp)
+	err = vdt.Set(val)
+	if err != nil {
+		return fmt.Errorf("setting varying data type value: %w", err)
+	}
+	*vnp = VersionedNeighbourPacket(vdt)
+	return nil
+}
+
+// Value returns the current VDT value
+func (vnp *VersionedNeighbourPacket) Value() (val scale.VaryingDataTypeValue) {
+	vdt := scale.VaryingDataType(*vnp)
+	return vdt.Value()
+}
+
+// NeighbourPacketV1 represents a network-level neighbour message
+// currently, round and setID represents a struct containing an u64
+// https://github.com/paritytech/substrate/blob/master/client/finality-grandpa/src/communication/mod.rs#L660
+type NeighbourPacketV1 struct {
```

Comment:
Is there a V2?

Reply:
Currently not, but as in substrate they use the 'V1' suffix I believe will exists another version in the future